frida-gum/src/range_details.rs

```diff
@@ -72,6 +105,40 @@ impl<'a> RangeDetails<'a> {
         }
     }
 
+    /// Get a [`RangeDetails`] for the range containing the given address.
+    pub fn with_address(address: u64) -> Option<RangeDetails<'a>> {
+        let mut context = SaveRangeDetailsByAddressContext {
+            address,
+            details: std::ptr::null_mut(),
+        };
+        unsafe {
+            gum_sys::gum_process_enumerate_ranges(
+                gum_sys::_GumPageProtection_GUM_PAGE_NO_ACCESS,
+                Some(save_range_details_by_address),
+                &mut context as *mut _ as *mut c_void,
+            );
+        }
+
+        if !context.details.is_null() {
+            Some(RangeDetails::from_raw(context.details))
+        } else {
+            None
+        }
+    }
+
+    pub fn enumerate_with_prot(
```

Comment:
Please add documentation for this function. I'll have to get around to `#![deny(missing_docs)]`...

Reply:
Ok. Will do...

Reply:
Done.